contracts/protocol/v1/traders/P1InverseOrders.sol

```diff
@@ -0,0 +1,495 @@
+    using SafeMath for uint256;
+
+    // ============ Constants ============
+
+    // EIP191 header for EIP712 prefix
+    bytes2 constant private EIP191_HEADER = 0x1901;
+
+    // EIP712 Domain Name value
+    string constant private EIP712_DOMAIN_NAME = "P1InverseOrders";
+
+    // EIP712 Domain Version value
+    string constant private EIP712_DOMAIN_VERSION = "1.0";
+
+    // Hash of the EIP712 Domain Separator Schema
+    /* solium-disable-next-line indentation */
+    bytes32 constant private EIP712_DOMAIN_SEPARATOR_SCHEMA_HASH = keccak256(abi.encodePacked(
+        "EIP712Domain(",
+        "string name,",
+        "string version,",
+        "uint256 chainId,",
+        "address verifyingContract",
+        ")"
+    ));
+
+    // Hash of the EIP712 LimitOrder struct
+    /* solium-disable-next-line indentation */
+    bytes32 constant private EIP712_ORDER_STRUCT_SCHEMA_HASH = keccak256(abi.encodePacked(
+        "Order(",
+        "bytes32 flags,",
+        "uint256 amount,",
+        "uint256 limitPrice,",
+        "uint256 triggerPrice,",
+        "uint256 limitFee,",
+        "address maker,",
+        "address taker,",
+        "uint256 expiration",
+        ")"
+    ));
+
+    // Bitmasks for the flags field
+    bytes32 constant FLAG_MASK_NULL = bytes32(uint256(0));
+    bytes32 constant FLAG_MASK_IS_BUY = bytes32(uint256(1));
+    bytes32 constant FLAG_MASK_IS_DECREASE_ONLY = bytes32(uint256(1 << 1));
+    bytes32 constant FLAG_MASK_IS_NEGATIVE_LIMIT_FEE = bytes32(uint256(1 << 2));
+
+    // ============ Enums ============
+
+    enum OrderStatus {
+        Open,
+        Approved,
+        Canceled
+    }
+
+    // ============ Structs ============
+
+    struct Order {
+        bytes32 flags;
+        uint256 amount;
+        uint256 limitPrice;
+        uint256 triggerPrice;
+        uint256 limitFee;
+        address maker;
+        address taker;
+        uint256 expiration;
+    }
+
+    struct Fill {
+        uint256 amount;
+        uint256 price;
+        uint256 fee;
+        bool isNegativeFee;
+    }
+
+    struct TradeData {
+        Order order;
+        Fill fill;
+        TypedSignature.Signature signature;
+    }
+
+    struct OrderQueryOutput {
+        OrderStatus status;
+        uint256 filledAmount;
+    }
+
+    // ============ Events ============
+
+    event LogOrderCanceled(
+        address indexed maker,
+        bytes32 orderHash
+    );
+
+    event LogOrderApproved(
+        address indexed maker,
+        bytes32 orderHash
+    );
+
+    event LogOrderFilled(
+        bytes32 orderHash,
+        bytes32 flags,
+        uint256 triggerPrice,
+        Fill fill
+    );
+
+    // ============ Immutable Storage ============
+
+    // address of the perpetual contract
+    address public _PERPETUAL_V1_;
+
+    // Hash of the EIP712 Domain Separator data
+    bytes32 public _EIP712_DOMAIN_HASH_;
+
+    // ============ Mutable Storage ============
+
+    // order hash => filled amount (in base currency)
```

Comment:
the original says `// order hash => filled amount (in position amount)` and I think thats fine to maintain here as well

Reply:
yeah, “base” was wrong